### PR TITLE
Issue #15214: migrated indentation sections to follow whole config testing

### PR DESCRIPTION
**Pull request by Zopsss**
#15214

  | 4.2 Block indentation: +2 spaces | Indentation
-- | -- | --

4.2 Block indentation: +2 spaces 	[Indentation](https://checkstyle.org/checks/misc/indentation.html#Indentation)

### CODE REVIEW

**Comment by Zopsss** [comment]
@rdiachenko, referring to: https://github.com/checkstyle/checkstyle/pull/15332#pullrequestreview-2194750316

this PR is comparatively small to the last "indentation" PR.  For now, I have only migrated section 4.2, the other 2 sections are still remaining: 4.5.2 & 4.8.4.1

should I migrated them in this single PR or should I send different PRs for each section? though these PRs will be exactly same as all these sections have same input files.

**Comment by rdiachenko** [comment]
> @rdiachenko, referring to: [#15332 (review)](https://github.com/checkstyle/checkstyle/pull/15332#pullrequestreview-2194750316)
> 
> this PR is comparatively small to the last "indentation" PR. For now, I have only migrated section 4.2, the other 2 sections are still remaining: 4.5.2 & 4.8.4.1
> 
> should I migrated them in this single PR or should I send different PRs for each section? though these PRs will be exactly same as all these sections have same input files.

I'm good with the size of the current PR. Please send a PR per section.

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule42blockindentation/InputIndentationCorrect.java [hunk 1]
```diff
@@ -99,5 +102,9 @@ public String trimTrailingFrom(CharSequence sequence) {
         };
 
     private static final String ZEROES = "0\u0660\u06f0" + "\u0c66\u0ce6" + "\u1c40\u1c50";
+    // 3 violations above:
+    //  'Unicode escape(s) usage should be avoided.'
+    //  'Unicode escape(s) usage should be avoided.'
+    //  'Unicode escape(s) usage should be avoided.'
```
Comment (rdiachenko) [inline src/it/resources/com/google/checkstyle/test/chapter4formatting/rule42blockindentation/InputIndentationCorrect.java:108]: How these violations are related to the current section?

Comment (Zopsss) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule42blockindentation/InputIndentationCorrect.java:108]: My bad, they aren't related. I accidentally kept them. Should I remove the whole variables or just fix the violations? I don't think the variable is used to cover any edge cases.

**Comment by Zopsss** [comment]
> I'm good with the size of the current PR. Please send a PR per section.

Okay, let's get this PR approved first. After that I'll send similar PRs for the other sections

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule42blockindentation/InputIndentationCorrect.java (hunk 1 above)
Comment (romani) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule42blockindentation/InputIndentationCorrect.java:108]: there is no Indentation related content just make 
`private static final String ZEROES = "" + "";`

**Review by romani — APPROVED** [review]
I am ok to merge.

point above should be fixed.

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule42blockindentation/InputIndentationCorrect.java (hunk 1 above)
Comment (Zopsss) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule42blockindentation/InputIndentationCorrect.java]: done

File: src/it/java/com/google/checkstyle/test/chapter4formatting/rule42blockindentation/BlockIndentation2SpacesTest.java [hunk 2]
```diff
@@ -25,84 +25,68 @@
...
-        verifyWithConfigParser(MODULES, filePath);
+        verifyWithWholeConfig(getPath("InputIndentationCorrectAnnotationArrayInit.java"));
     }
 
     @Test
-    public void tetsFastMatcher() throws Exception {
-        final String filePath = getPath("InputFastMatcher.java");
-        verifyWithConfigParser(MODULES, filePath);
+    public void testFastMatcher() throws Exception {
+        verifyWithWholeConfig(getPath("InputFastMatcher.java"));
```
Comment (Zopsss) [inline src/it/java/com/google/checkstyle/test/chapter4formatting/rule42blockindentation/BlockIndentation2SpacesTest.java:90]: corrected method name

**Review by rdiachenko — APPROVED** [review]
lgtm

**Pull request merged**